src/audio_sdl.cpp

```diff
@@ -99,9 +125,27 @@ SdlAudio::SdlAudio() :
 #else
 	SDL_PauseAudio(0);
 #endif
+
+	// Start midiout polling thread
+	// TODO: don't make a thread if we're on a platform that won't use MidiOut
+	midiout_mutex = SDL_CreateMutex();
+	midiout_thread = SDL_CreateThread(MidioutThreadMain, "MidioutThread", this);
```

Comment:
Why not `std::thread` and `std::mutex`? Then we avoid sdl version issues and use standard library facilities.

Reply:
I do love me some `std::thread` and `std::mutex` but I've been told it's less portable, some platforms don't have good implementations of it. My original thread implementation used it, but doing it this way keeps the threading inside SDL, though at the expense of a bit of ifdef hell

Reply:
Let's re-evaluate, even if `std::thread` is not well supported what about `std::mutex`? The `lock_guard` and `unique_lock` RAII facilities are really great to ensure no deadlocks from leaks.

Reply:
It occurs to me also that other than using the thread API, there is nothing SDL specific in this entire PR. I wonder if the management of this thread even belongs here?